keps/sig-node/0009-node-heartbeat.md

```diff
@@ -189,27 +189,23 @@ Once all the code changes are done, we will:
    but it’s fine to reduce its frequency (though we should continue writing it at
    least once per eviction period).
 
-
-To be considered:
-
-1. We may consider reducing frequency of NodeStatus updates to once every 5 minutes
-   (instead of 1 minute). That would help with performance/scalability even more.
-   Caveats:
-   - NodeProblemDetector is currently updating (some) node conditions every 1 minute
-     (unconditionally, because lastHeartbeatTime always changes). To make reduction
-     of NodeStatus updates frequency really useful, we should also change NPD to
-     work in a similar mode (check periodically if condition changes, but report only
-     when something changed or no status was reported for a given time) and decrease
-     its reporting frequency too.
-   - In general, we recommend to keep frequencies of NodeStatus reporting in both
-     Kubelet and NodeProblemDetector in sync (once all changes will be done) and
-     that should be reflected in [NPD documentation][].
-   - Note that reducing frequency to 1 minute already gives us almost 6x improvement.
-     It seems more than enough for any foreseeable future assuming we won’t
-     significantly increase the size of object Node.
-     Note that if we keep adding node conditions owned by other components, the
-     number of writes of Node object will go up. But that issue is separate from
-     that proposal.
+Based on experiments, unsuccessful attempt to enable `Server Side Apply` due to
+performance issues it was causing, in v1.17 release, we will:
+- reduce frequency of NodeStatus update to once every 5 minutes instead of
+  1 minute (note that if any condition changes, the update will happen within
+  next 10 seconds anyway, so we are only talking about periodic heartbeats)
+- reduce frequency of updating NodeStatuses in NodeProblemDetector also to
```

Comment:
I assume that if the status changes, NPD would also make the change promptly

Reply:
Yes - clarified in the text.